src/modules/job-manager/jobtap.c

```diff
@@ -870,35 +871,79 @@ static zlistx_t *path_list (const char *searchpath, const char *path)
     return NULL;
 }
 
+static int plugin_set_name (flux_plugin_t *p,
+                            const char *basename)
+{
+    int rc = -1;
+    char *q;
+    char *copy = NULL;
+    const char *name = flux_plugin_get_name (p);
+
+    /*  It is ok to have a custom name, but that name may
+     *   not contain '/' or '.'
+     */
+    if (name && !strchr (name, '/') && !strchr (name, '.'))
+        return 0;
+    if (!(copy = strdup (basename)))
+        return -1;
+    if ((q = strchr (copy, '.')))
+        *q = '\0';
+    rc = flux_plugin_set_name (p, copy);
+    free (copy);
```

Comment:
I tried to avoid returning errno-only reponses and include a useful error message in all responses instead. Not sure an errno from plugin_set_name() will be useful to the caller, but I'll try to preserve it anyway.

Reply:
Yeah, I guess the value of errno doesn't matter that much if there's a human readable error.  We just need to ensure errno != 0 or `flux_respond_error()` will fail and not send the response at all.  (In case that wasn't clear)

Reply:
Ah, good point! I didn't know that or had forgotten it. I wonder if that is overall desired behavior now that `flux_respond_error()` can take an error message?

Reply:
Regardless, I'll try to ensure errno is preserved here

Reply:
I struggled with that one in that now-defunct EFLUXED PR, because in some cases you don't know if the caller will do the extra work to decode a human readable error, so returning errnum=0 or a misleading value is not great and should be detected by unit tests.

In this case the RPC is not really intended to be used by anything other than the plugin loader, so you could just call `flux_respond_error (h, msg, EINVAL, errtxt)` and just make sure errtxt is non-NULL?

Reply:
Ah, good point. I'll try to preserve errno but also do something like this?
```C
  flux_respond_error (h, msg, errno != 0 ? errno : EINVAL, errmsg)
```

Reply:
Yeah that might be fine in this case.